crates/bevy_render/src/mesh/mod.rs

```diff
@@ -27,5 +33,47 @@ impl Plugin for MeshPlugin {
             .register_type::<Vec<Entity>>()
             // 'Mesh' must be prepared after 'Image' as meshes rely on the morph target image being ready
             .add_plugins(RenderAssetPlugin::<Mesh, Image>::default());
+
+        let Ok(render_app) = app.get_sub_app_mut(RenderApp) else {
+            return;
+        };
+
+        render_app.init_resource::<MeshVertexBufferLayouts>();
+    }
+}
+
+#[derive(Clone, Debug)]
+pub struct MeshVertexBufferLayoutRef(pub Arc<MeshVertexBufferLayout>);
+
+#[derive(Clone, Default, Resource)]
+pub struct MeshVertexBufferLayouts(HashSet<MeshVertexBufferLayoutRef>);
```

Comment:
Does this need to be pub? For all intents and purposes, this is an implementation detail.

Reply:
It does because otherwise `rustc` complains that it's being used in a public interface (specifically, `<Mesh as RenderAsset>::Param`).